### DISCUSSION TITLE
V7: reset function now trigger form's onReset event

### DISCUSSION THREAD

**Discussion by rexebin** [react-hook-form/react-hook-form#4692] [General]
I have been using RHF, a wonderful library, thank you for your hard work!
Very excited to update to V7, found out calling `reset` method now trigger form's `onReset` event, it did not do that prior v7. Is this behaviour intended? 

I have this lazy design to using form as detail page as well(disabled form element with styling), so when I hit `cancel` button, I set form element to disabled doubling as a detailed page. 

I some times I need to something extra when I hit reset button, for example, if the entity is new, hitting the `cancel` button triggering `onReset` event, I want to move away from the `new` route or close a dialog, because an empty entity doesn't have a detail page.

Now the RHF `reset` function trigger `onReset` event, making the above situation a bit tricky. 

Is there an option on `reset` method for disabling the trigger of `onReset` event?

**Comment by jorisre** [comment]
Thanks for reporting, can you share a codesandbox with youyr problem ? It helps us to investigate

**Reply by rexebin** [reply]
https://codesandbox.io/s/dreamy-sea-1ft4e?file=/src/App.tsx

I got a regular reset button and a button to call the `reset` function. 

1. make sure the reset message is cleared
2. click `call reset()` button sets off the `onReset` event, the reset message is shown.

or observe the console logs.

**Reply by bluebill1049** [reply]
are you missing type="Button"

**Reply by rexebin** [reply]
@bluebill1049 I added `type="button"`, same behaviour.  Calling `reset` triggers `onReset` event.

**Reply by rexebin** [reply]
help please

**Reply by bluebill1049** [reply]
We do reset the form when invoke reset API to clear the input value. 
<img width="563" alt="Screen Shot 2021-04-17 at 5 55 27 pm" src="https://user-images.githubusercontent.com/10513364/115106086-1aec1880-9fa6-11eb-8624-c7290430da99.png">

**Reply by bluebill1049** [reply]
it's always been there since the very first version of hook form.

**Reply by rexebin** [reply]
Thank you! I have worked my way around it. Somehow it's only became obvious when I upgrade to v7 from v6. But it does make sense to reset the form.

**Reply by bluebill1049** [reply]
after you invoke the `reset` api, why shouldn't we reset the form? I am happy to hear your reasoning to adjust if requires.

**Reply by rexebin** [reply]
As I explained at the very beginning, I had some logic in the onReset event. I had buttons outside of the form triggering the onReset event, I would route to list view if the form is read only or is editing a brand new entity(cancel creating new entity); I would set the form to read only mode(detail view) if the form is in edit mode with a existing entity.

Actually I should not do all above in one onReset event. V7's triggering of onReset event actually helped me and forced me to separate my logic, making the code a lot cleaner. 

So I don't see a reason why RHF should change its reset behaviour. The reset function and onReset event should be a single purpose event, that is to reset the form.
